cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactClassLoaderFactory.java

```diff
@@ -75,7 +73,8 @@ CloseableClassLoader createClassLoader(Location artifactLocation) throws IOExcep
       // TODO: CDAP-5613. We shouldn't always expose the Spark classes.
       programRunner = programRunnerFactory.create(ProgramType.SPARK);
       if (programRunner instanceof ProgramClassLoaderProvider) {
-        programClassLoader = ((ProgramClassLoaderProvider) programRunner).createProgramClassLoader(cConf, unpackDir);
+        programClassLoader = new ProgramClassLoader(
+          cConf, unpackDir, ((ProgramClassLoaderProvider) programRunner).createProgramClassLoaderParent());
       }
```

Comment:
The exception caught in line 81 is actually thrown by the `programRunnerFactory.create(ProgramType.SPARK)` call when Spark is not supported in the cluster.

If the `programRunner` is not a `ProgramClassLoaderProvider`, then the default program classloader will be used for the artifact inspection.

I've moved the `if` block outside of the `try` block to make it clearer.